src/components/Header.jsx

```diff
@@ -103,54 +122,71 @@ function Header() {
   const [anchorEl, setAnchorEl] = useState(null);
 
   return (
-    <AppBar position="static" color="secondary">
-      <UnderConstructionAlert t={t} />
-      <OnRenderSnackbar />
-      <Toolbar>
-        <MuiLink
-          component={Link}
-          to="/"
-          variant="h5"
-          color="primary"
-          underline="none"
-          sx={{ flexGrow: 1 /** Take up remaining space */ }}>
-          {BRAND_NAME.toUpperCase()}
-        </MuiLink>
-        <Hidden smDown>
-          <HeaderNavMenu links={links} />
+    <HideOnScroll>
+      <AppBar
+        color="secondary"
+        sx={{ zIndex: (theme) => theme.zIndex.drawer + 1 }}>
+        <UnderConstructionAlert t={t} />
+        <Toolbar>
+          <OnRenderSnackbar />
+          <MuiLink
+            component={Link}
+            to="/"
+            variant="h5"
+            color="primary"
+            underline="none"
+            sx={{ flexGrow: 1 /** Take up remaining space */ }}>
+            {BRAND_NAME.toUpperCase()}
+          </MuiLink>
+          <Hidden smDown>
+            <HeaderNavMenu links={links} />
+          </Hidden>
+          <IconButton
+            color="primary"
+            onClick={(e) => setAnchorEl(e.currentTarget)}
+            sx={{ px: 2 }}>
+            <LanguageIcon />
+          </IconButton>
+          <AuthGrowButton t={t} />
+        </Toolbar>
+        <Hidden smUp>
+          <Toolbar variant="dense">
+            <HeaderNavMenu links={links} />
+          </Toolbar>
         </Hidden>
-        <IconButton
-          color="primary"
-          onClick={(e) => setAnchorEl(e.currentTarget)}
-          sx={{ px: 2 }}>
-          <LanguageIcon />
-        </IconButton>
-        <AuthGrowButton t={t} />
-      </Toolbar>
+        <Menu
+          open={Boolean(anchorEl)}
+          anchorEl={anchorEl}
+          onClose={() => setAnchorEl(null)}
+          anchorOrigin={{ horizontal: 'center', vertical: 'bottom' }}
+          transformOrigin={{ horizontal: 'center', vertical: 'top' }}>
+          {Object.entries(languages).map(([abbr, lang]) => (
+            <MenuItem
+              key={lang}
+              selected={abbr === i18n.language}
+              onClick={() => {
+                i18n.changeLanguage(abbr);
+                setAnchorEl(null);
+              }}>
+              {lang}
+            </MenuItem>
+          ))}
+        </Menu>
+      </AppBar>
+    </HideOnScroll>
+  );
+}
+
+export function HeaderSkeleton() {
```

Comment:
Imitates the header so that the side and main panels can be pushed down the right amount of space at the start.